From b4110a5f3977379a72571508cb75552bd868845e Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 22 Dec 2016 12:40:01 +0100
Subject: [PATCH 0676/1345] telephony: mvebu: sort TAL callbacks and interrupt
 handlers

commit  dd40e78236421f86bb56e492f8632add98fa567e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add sorting in order to obtain following file structure:
* /proc/tdm handling
* TAL callbacks (sorted)
* interrupt handling + tasklet callbacks
* a38x clock handling (in future it should to be moved to separate
  clock driver)
* driver routines (probe/remove/PM)

Change-Id: I9ff8a8f25b52439060578c3bc82e7e78deb8639c
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35045
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/telephony/mvebu_phone/mv_phone_dev.c |  140 +++++++++++++-------------
 1 file changed, 71 insertions(+), 69 deletions(-)

diff --git a/drivers/telephony/mvebu_phone/mv_phone_dev.c b/drivers/telephony/mvebu_phone/mv_phone_dev.c
index 8b92a5a..06c6060 100644
--- a/drivers/telephony/mvebu_phone/mv_phone_dev.c
+++ b/drivers/telephony/mvebu_phone/mv_phone_dev.c
@@ -498,6 +498,77 @@ void tdm_if_exit(void)
 	}
 }
 
+static int tdm_if_control(int cmd, void *arg)
+{
+	switch (cmd) {
+	case TDM_DEV_TDM_TEST_MODE_ENABLE:
+		test_enable = 1;
+		break;
+
+	case TDM_DEV_TDM_TEST_MODE_DISABLE:
+		test_enable = 0;
+		break;
+
+	default:
+		return -EINVAL;
+	};
+
+	return 0;
+}
+
+static int tdm2c_if_write(u8 *buffer, int size)
+{
+	if (test_enable)
+		return tdm2c_tx(buffer);
+
+	return 0;
+}
+
+static int tdmmc_if_write(u8 *buffer, int size)
+{
+	if (test_enable)
+		return tdmmc_tx(buffer);
+
+	return 0;
+}
+
+static void tdm_if_stats_get(struct tal_stats *tdm_if_stats)
+{
+	if (tdm_init == 0)
+		return;
+
+	tdm_if_stats->tdm_init = tdm_init;
+	tdm_if_stats->rx_miss = rx_miss;
+	tdm_if_stats->tx_miss = tx_miss;
+	tdm_if_stats->rx_over = rx_over;
+	tdm_if_stats->tx_under = tx_under;
+#ifdef CONFIG_MV_TDM_EXT_STATS
+	tdm2c_ext_stats_get(&tdm_if_stats->tdm_ext_stats);
+#endif
+}
+
+static struct tal_if tdm2c_if = {
+	.pcm_start	= tdm2c_if_pcm_start,
+	.pcm_stop	= tdm2c_if_pcm_stop,
+	.init		= tdm_if_init,
+	.exit		= tdm_if_exit,
+	.control	= tdm_if_control,
+	.write		= tdm2c_if_write,
+	.stats_get	= tdm_if_stats_get,
+};
+
+static struct tal_if tdmmc_if = {
+	.pcm_start	= tdmmc_if_pcm_start,
+	.pcm_stop	= tdmmc_if_pcm_stop,
+	.init		= tdm_if_init,
+	.exit		= tdm_if_exit,
+	.control	= tdm_if_control,
+	.write		= tdmmc_if_write,
+	.stats_get	= tdm_if_stats_get,
+};
+
+/* Interrupt handling and tasklet callbacks */
+
 /* Common interrupt top-half handler */
 static irqreturn_t tdm_if_isr(int irq, void *dev_id)
 {
@@ -765,75 +836,6 @@ static void tdm2c_if_reset_channels(unsigned long arg)
 	tdm2c_if_pcm_start();
 }
 
-static int tdm_if_control(int cmd, void *arg)
-{
-	switch (cmd) {
-	case TDM_DEV_TDM_TEST_MODE_ENABLE:
-		test_enable = 1;
-		break;
-
-	case TDM_DEV_TDM_TEST_MODE_DISABLE:
-		test_enable = 0;
-		break;
-
-	default:
-		return -EINVAL;
-	};
-
-	return 0;
-}
-
-static int tdm2c_if_write(u8 *buffer, int size)
-{
-	if (test_enable)
-		return tdm2c_tx(buffer);
-
-	return 0;
-}
-
-static int tdmmc_if_write(u8 *buffer, int size)
-{
-	if (test_enable)
-		return tdmmc_tx(buffer);
-
-	return 0;
-}
-
-static void tdm_if_stats_get(struct tal_stats *tdm_if_stats)
-{
-	if (tdm_init == 0)
-		return;
-
-	tdm_if_stats->tdm_init = tdm_init;
-	tdm_if_stats->rx_miss = rx_miss;
-	tdm_if_stats->tx_miss = tx_miss;
-	tdm_if_stats->rx_over = rx_over;
-	tdm_if_stats->tx_under = tx_under;
-#ifdef CONFIG_MV_TDM_EXT_STATS
-	tdm2c_ext_stats_get(&tdm_if_stats->tdm_ext_stats);
-#endif
-}
-
-static struct tal_if tdm2c_if = {
-	.init		= tdm_if_init,
-	.exit		= tdm_if_exit,
-	.pcm_start	= tdm2c_if_pcm_start,
-	.pcm_stop	= tdm2c_if_pcm_stop,
-	.control	= tdm_if_control,
-	.write		= tdm2c_if_write,
-	.stats_get	= tdm_if_stats_get,
-};
-
-static struct tal_if tdmmc_if = {
-	.init		= tdm_if_init,
-	.exit		= tdm_if_exit,
-	.pcm_start	= tdmmc_if_pcm_start,
-	.pcm_stop	= tdmmc_if_pcm_stop,
-	.control	= tdm_if_control,
-	.write		= tdmmc_if_write,
-	.stats_get	= tdm_if_stats_get,
-};
-
 /* Enable device interrupts. */
 void mv_phone_intr_enable(u8 dev_id)
 {
-- 
1.7.9.5

